core/asset/asset.go

```diff
@@ -28,20 +28,20 @@ type Repository interface {
 
 // Asset is a model that wraps arbitrary data with Compass' context
 type Asset struct {
-	ID          string                 `json:"id" diff:"-"`
-	URN         string                 `json:"urn" diff:"-"`
-	Type        Type                   `json:"type" diff:"-"`
-	Service     string                 `json:"service" diff:"-"`
-	Name        string                 `json:"name" diff:"name"`
-	Description string                 `json:"description" diff:"description"`
-	Data        map[string]interface{} `json:"data" diff:"data"`
-	URL         string                 `json:"url" diff:"url"`
-	Labels      map[string]string      `json:"labels" diff:"labels"`
+	ID          string                 `json:"id,omitempty" diff:"-"`
+	URN         string                 `json:"urn,omitempty" diff:"-"`
+	Type        Type                   `json:"type,omitempty" diff:"-"`
+	Service     string                 `json:"service,omitempty" diff:"-"`
+	Name        string                 `json:"name,omitempty" diff:"name"`
+	Description string                 `json:"description,omitempty" diff:"description"`
+	Data        map[string]interface{} `json:"data,omitempty" diff:"data"`
+	URL         string                 `json:"url,omitempty" diff:"url"`
+	Labels      map[string]string      `json:"labels,omitempty" diff:"labels"`
```

Comment:
Why is this change necessary? some of these fields should not be empty

Reply:
removed